File: Sovenire_Collenction_Backend/Models/Collection.cs
using System.ComponentModel.DataAnnotations;

namespace Souvenir_Collection_Backend.Models
{
    public class Collection
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        [MaxLength(150)]
        public string Title { get; set; } = string.Empty;

        [MaxLength(150)]
        public string Slug { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        [MaxLength(50)]
        public string Type { get; set; } = string.Empty;

        public string Image { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [System.Text.Json.Serialization.JsonIgnore]
        public ICollection<Product> Products { get; set; } = new List<Product>();
    }
}

File: Souvenir_Collection_Frontend/lib/models/collection.dart
class Collection {
  final String id;
  final String title;
  final String slug;
  final String description;
  final String type;
  final String image;
  final DateTime? createdAt;

  const Collection({
    required this.id,
    required this.title,
    this.slug = '',
    this.description = '',
    this.type = '',
    this.image = '',
    this.createdAt,
  });

  factory Collection.fromJson(Map<String, dynamic> json) {
    return Collection(
      id: json['id'] as String? ?? '',
      title: json['title'] as String? ?? '',
      slug: json['slug'] as String? ?? '',
      description: json['description'] as String? ?? '',
      type: json['type'] as String? ?? '',
      image: json['image'] as String? ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'title': title,
        'slug': slug,
        'description': description,
        'type': type,
        'image': image,
        'createdAt': createdAt?.toIso8601String(),
      };
}
